vendor/bat-native-ads/src/bat/ads/internal/ad_serving/ad_notifications/ad_notification_serving.cc

```diff
@@ -174,7 +174,7 @@ void AdServing::MaybeServeAdForParentChildCategories(
     const CreativeAdNotificationList eligible_ads =
         eligible_ad_notifications.Get(ads,
             last_delivered_creative_ad_, ad_events);
-
+    ads_client_->SetIntegerPref(prefs::kEligibleAdsCount, eligible_ads.size());
```

Comment:
We should not be storing diagnostics under a pref, we should have a diagnostics object, which has a data structure and this data structure is serialized, i.e. `AdDiagnostics()->Get()->SetEligibleAdCount(...)`. Then `AdsImpl::GetInternalsInfo`  can get the serialized JSON from this object